File: 1161-maximum-level-sum-of-a-binary-tree/1161-maximum-level-sum-of-a-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int maxLevelSum(TreeNode* root) {
        //BFS
        vector<int> sum_vec;
        queue<TreeNode*> q;

        q.push(root);
        while(!q.empty()){
            int sum = 0;
            int n = q.size();
            for(int i=0; i<n; i++){
                TreeNode* current = q.front();
                q.pop();
                if(current->left) q.push(current->left);
                if(current->right) q.push(current->right);
                sum += current->val;
            }
            sum_vec.push_back(sum);
        }

        int result, max = -INT_MAX;
        for(int i=0; i<sum_vec.size(); i++){
            if(max < sum_vec[i]){
                max = sum_vec[i];
                result = i;
            }
        }

        return result+1;
    }
};

File: 1161-maximum-level-sum-of-a-binary-tree/1161-maximum-level-sum-of-a-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def maxLevelSum(self, root: Optional[TreeNode]) -> int:
        if not root:
            return 0

        q = deque()
        q.append(root)
        maxi = float('-inf')
        level = 1
        ans = 0

        while q:
            total = 0
            n = len(q)
            
            for _ in range(n):
                current = q.popleft()
                total += current.val
                if current.left:
                    q.append(current.left)
                if current.right:
                    q.append(current.right)

            if maxi < total:
                maxi = total
                ans = level
            level += 1
    
        return ans